dev-lang/fpc/fpc-3.2.2.ebuild

```diff
@@ -0,0 +1,160 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit prefix toolchain-funcs
+
+HOMEPAGE="https://www.freepascal.org/"
+DESCRIPTION="Free Pascal Compiler"
+SRC_URI="mirror://sourceforge/freepascal/fpcbuild-${PV}.tar.gz
+	mirror://sourceforge/freepascal/fpc-${PV}.source.tar.gz
+	amd64? ( mirror://sourceforge/freepascal/${P}.x86_64-linux.tar )
+	x86? ( mirror://sourceforge/freepascal/${P}.i386-linux.tar )
+	doc? ( mirror://sourceforge/freepascal/Documentation/${PV}/doc-html.tar.gz -> ${P}-doc-html.tar.gz )
+"
+
+SLOT="0"
+LICENSE="GPL-2 LGPL-2.1-with-linking-exception"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc source"
+
+RESTRICT="strip" #269221
+
+S="${WORKDIR}/fpcbuild-${PV}/fpcsrc"
+
+pkg_pretend() {
+	if [[ "${MERGE_TYPE}" != binary ]]; then
+		# Bug 475210
+		if "$(tc-getLD)" --version | grep -q "GNU gold"; then
+			eerror "fpc does not function correctly when built with the gold linker."
+			eerror "Please select the bfd linker with binutils-config."
+			die "GNU gold detected"
+		fi
+	fi
+}
+
+src_unpack() {
+	case "${ARCH}" in
+		amd64)	FPC_ARCH="x86_64"    PV_BIN="${PV}" ;;
+		x86)	FPC_ARCH="i386"      PV_BIN="${PV}" ;;
+		*)	die "This ebuild doesn't support ${ARCH}." ;;
+	esac
+
+	unpack ${A}
+
+	tar -xf "${PN}-${PV_BIN}.${FPC_ARCH}-linux/binary.${FPC_ARCH}-linux.tar" || die "Unpacking binary.${FPC_ARCH}-linux.tar failed!"
+	tar -xzf "base.${FPC_ARCH}-linux.tar.gz" || die "Unpacking base.${FPC_ARCH}-linux.tar.gz failed!"
```

Comment:
Can't these use `unpack ./file.tar.gz` instead of tar?

Reply:
Replaced by unpack.